src/whenever/__init__.py

```diff
@@ -3499,9 +3579,14 @@ def canonical_format(self, sep: Literal[" ", "T"] = "T") -> str:
 
     @classmethod
     def from_canonical_format(cls, s: str, /) -> LocalSystemDateTime:
-        if not _match_offset_str(s):
-            raise InvalidFormat()
-        return cls._from_py_unchecked(_fromisoformat(s))
+        try:
+            if not _match_offset_str(s):
+                raise ValueError("Input seems malformed")
```

Comment:
Let's stick to one generic message (i.e. `f"Could not parse as canonical format string: {s!r}"`)